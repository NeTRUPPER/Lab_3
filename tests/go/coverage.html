
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>queue_test: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">queue_test/queue.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package main

import (
        "fmt"
)

// Node represents a single element in the queue.
type Node struct {
        value interface{}
        next  *Node
}

// Queue represents a FIFO (first-in, first-out) queue implemented using a linked list.
type Queue struct {
        head *Node
        tail *Node
        size int
}

// New creates a new empty queue.
func NewQueue() *Queue <span class="cov8" title="1">{
        return &amp;Queue{
                head: nil,
                tail: nil,
                size: 0,
        }
}</span>

// Push adds an element to the end of the queue.
func (q *Queue) Push(item interface{}) <span class="cov8" title="1">{
        newNode := &amp;Node{value: item, next: nil}
        if q.tail == nil </span><span class="cov8" title="1">{
                q.head = newNode
                q.tail = newNode
        }</span> else<span class="cov8" title="1"> {
                q.tail.next = newNode
                q.tail = newNode
        }</span>
        <span class="cov8" title="1">q.size++</span>
}

// Pop removes and returns the element at the front of the queue.
// Returns an error if the queue is empty.
func (q *Queue) Pop() (interface{}, error) <span class="cov8" title="1">{
        if q.IsEmpty() </span><span class="cov8" title="1">{
                return nil, fmt.Errorf("queue is empty")
        }</span>
        <span class="cov8" title="1">elem := q.head.value
        q.head = q.head.next
        if q.head == nil </span><span class="cov8" title="1">{
                q.tail = nil
        }</span>
        <span class="cov8" title="1">q.size--
        return elem, nil</span>
}

// Peek returns the element at the front of the queue without removing it.
// Returns an error if the queue is empty.
func (q *Queue) Peek() (interface{}, error) <span class="cov8" title="1">{
        if q.IsEmpty() </span><span class="cov8" title="1">{
                return nil, fmt.Errorf("queue is empty")
        }</span>
        <span class="cov8" title="1">return q.head.value, nil</span>
}

// Size returns the number of elements in the queue.
func (q *Queue) Size() int <span class="cov8" title="1">{
        return q.size
}</span>

// IsEmpty checks if the queue is empty.
func (q *Queue) IsEmpty() bool <span class="cov8" title="1">{
        return q.size == 0
}</span>
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
