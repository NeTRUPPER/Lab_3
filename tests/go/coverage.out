mode: set
queue_test/queue.go:21.24,27.2 1 1
queue_test/queue.go:30.40,32.19 2 1
queue_test/queue.go:32.19,35.3 2 1
queue_test/queue.go:35.8,38.3 2 1
queue_test/queue.go:39.2,39.10 1 1
queue_test/queue.go:44.44,45.17 1 1
queue_test/queue.go:45.17,47.3 1 1
queue_test/queue.go:48.2,50.19 3 1
queue_test/queue.go:50.19,52.3 1 1
queue_test/queue.go:53.2,54.18 2 1
queue_test/queue.go:59.45,60.17 1 1
queue_test/queue.go:60.17,62.3 1 1
queue_test/queue.go:63.2,63.26 1 1
queue_test/queue.go:67.28,69.2 1 1
queue_test/queue.go:72.32,74.2 1 1
